### PR TITLE
Makefile.dep: sort dependencies

### PR DESCRIPTION
**Pull request by kfessel**
While searching for dependancies from core to sys i sorted Makefile.dep (and found non) this patch contains the sorted Makefile.deps

### Contribution description

move most dependencies from <RIOTBASE>/Makefile.dep to the Makefile.dep where
dependent module resides

### Testing procedure

this needs build tests maybe murdock has some time

### CODE REVIEW

File: drivers/Makefile.dep [hunk 1]
```diff
@@ -140,3 +140,37 @@ endif
...
+
+ifneq (,$(filter saul_adc,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_adc
+endif
+
+ifneq (,$(filter saul_gpio,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_gpio
+endif
+
+ifneq (,$(filter saul,$(USEMODULE)))
```
Comment (aabadie) [inline drivers/Makefile.dep:156]: This appears twice

**Comment by aabadie** [comment]
Thanks for this PR. This is something I wanted to do for a long time.

There are checks in the build system to verify the list of modules included in each build. This will to detect that this PR doesn't add any regression.

File: drivers/Makefile.dep (hunk 1 above)
Comment (kfessel) [reply drivers/Makefile.dep:156]: yes it was there twice before, and i moved both, but i also think one off them is superfluous

**Comment by kfessel** [comment]
I am quiet sure that murdock is good at handling such change.

**Comment by aabadie** [comment]
> I am quiet sure that murdock is good at handling such change.

Yes it's good sign when Murdock is green but modifying code related to a core feature of the build system (the dependency resolution) requires very special care. Sometimes things could be missed although Murdock is green.

**Comment by kfessel** [comment]
atm:murdock fails with some errors that don't seem to be related to this pr: on is a connection error of git,
the other is some failure in prng selection Makefile.dep says always add the default to the list, murdock says not more than one (this should always fail if someone does not select the default but it does not i just build test/pkg_relic (for native since i have no esp compiler))

**Comment by kfessel** [comment]
> ... Sometimes things could be missed although Murdock is green.
Yes but at least it builds all tests with much more tool chains than i ever could, and since this is mostly about building and not about the inner workings of the OS i think Murdock is a realy good check

**Comment by kfessel** [comment]
rebased to solve master confilict

**Comment by fjmolinas** [comment]
> @aabadie: is there some kind of beta branch? like an open merge test

Not sure what you mean by this, care to elaborate?

**Comment by fjmolinas** [comment]
I'm running some more extensive tests, will have the results in a couple hours.

**Comment by kfessel** [comment]
> > @aabadie: is there some kind of beta branch? like an open merge test
> 
> Not sure what you mean by this, care to elaborate?

Like a branch like master, which merges changes (PRs) from different directions but moving more. Like my local develop-branch but common for RIOT. But i guess master is this branch and the less moving branches are the rc/release branches

**Comment by fjmolinas** [comment]
> Like a branch like master, which merges changes (PRs) from different directions but moving more. Like my local develop-branch but common for RIOT. But i guess master is this branch and the less moving branches are the rc/release branches.

Ah yes, that might be confusing, but its just as you say master is the develop branch in a way, and the stable branches are the release branches.

**Comment by kfessel** [comment]
@aabadie: you set the label `need sqashing`: Should I rebase and squash?

how are the results of your `more extensive tests`?

**Comment by fjmolinas** [comment]
I ran `ave_all_dependencies_resolution_variables.sh` against master and this PR, there is a big diff because some modules get included many more times with this PR, but the sorted version is still the same. So all OK, here is the [diff](https://drive.google.com/file/d/12dgbh6tJG68EOsZMOzJZ5EeXxRMZFCAr/view?usp=sharing)

**Comment by fjmolinas** [comment]
> @aabadie: you set the label need sqashing: Should I rebase and squash?

No need to rebase just squashing.

**Comment by kfessel** [comment]
squashed
 
i think what is left for the sorting will sort out in future, this is a good start

File: cpu/native/Makefile.dep [hunk 2]
```diff
@@ -21,6 +21,10 @@ ifneq (,$(filter periph_rtc,$(USEMODULE)))
   USEMODULE += xtimer
 endif
 
+ifneq (,$(filter can_linux,$(USEMODULE)))
+  LINKFLAGS += -lsocketcan
+endif
```
Comment (aabadie) [inline cpu/native/Makefile.dep:26]: Normally, this should go to `cpu/native/Makefile.include`, but I won't insist.

Comment (kfessel) [reply cpu/native/Makefile.dep:26]: I will move that, but I think we should get this into master, to get the seperation in and the majority of this cleanup done, i think there will be future smaller patches that move such 3 line chunks at even more correct places.

**Comment by kfessel** [comment]
moved the socket_can include

- I think we should get this into master, to get the separation in and the majority of this cleanup done
- i think there will be future smaller patches that move 3 line chunks at even more correct places

**Comment by kfessel** [comment]
rebased to solve the conflicts (followed all change to relevant makefiles in master)

**Comment by kfessel** [comment]
rebased to solve the conflicts (followed all change to Makefiles(.dep) in master)

could not build test since BOARD=native is not whitelisted for tests/ieee802154_submac and i got not whitelisted board

**Comment by kfessel** [comment]
Hi, @aabadie do you want to approve this PR

**Comment by kfessel** [comment]
@fjmolinas: since you did test this patch would you also approve it?

**Comment by fjmolinas** [comment]
@kfessel I'm fine with the changes, I'll ACK and merge after Hard Feature Freeze next monday!

**Review by fjmolinas — APPROVED** [review]
ACK, @kfessel please can you rebase one last time!

**Comment by kfessel** [comment]
rebased to solve the conflicts (followed all change to Makefile(.dep) in master)

**Comment by fjmolinas** [comment]
Go! thanks for the cleanup and the patience @kfessel!

**Pull request merged**